src/effects/Fire.js

```diff
@@ -129,6 +137,33 @@ export default class Effect {
       // Initialize the fire-preset dropdown.
       Effect._createFirePresets(dialog);
     }
+
+    // enables and disables the color buttons
+    function EnableDisableColorButtons(dialog, state) {
+
+      for (let i = 1; i <= 5; i++) {
+        dialog.getBuilder().get_object('fire-color-' + i).set_sensitive(!state);
+      }
+    }
+
+    const switchWidget = dialog.getBuilder().get_object('fire-random-color');
+    if (switchWidget) {
+      // Connect to the "state-set" signal to update preferences dynamically based on
+      // the switch state.
+      switchWidget.connect('state-set', (widget, state) => {
+        EnableDisableColorButtons(dialog,
```

Comment:
Just for consistency.

```suggestion
        enableDisableColorButtons(dialog,
```